core/domain/state_domain.py

```diff
@@ -979,6 +1872,23 @@ def validate(
             raise utils.ValidationError(
                 'Hint(s) must be specified if solution is specified')
 
+        # TODO: Find a way to encode these checks more declaratively.
```

Comment:
@lkbhitesh07 Every TODO should have an issue number, and that issue should be described really clearly and in detail.

Reply:
Done!